adoc/chapters/programming_interface.adoc

```diff
@@ -482,77 +503,104 @@ special member functions and member functions in order to fulfill the copy,
-[width="100%",options="header",separator="@",cols="2,1"]
-|====
-@ Special member function _(see <<rule-of-five>> and <<rule-of-zero>>)_ @ Description
-a@
-[source]
+[[sec:value-semantics-special-members]]
+==== Special member functions
+
+This section describes the special member functions that may be provided by
+classes providing the common by-value semantics.
+If a class provides any one of these special member functions, all of the
+special member functions must be provided (see also the <<rule-of-five>> and the
+<<rule-of-zero>>).
+Each of these classes may provide further member functions, static member
+functions, etc., as documented in their respective sections.
+
+'''
+
+.[apititle]#Copy constructor#
+[source,role=synopsis,id=api:value-semantics-ctor-copy]
 ----
-T(const T& rhs);
+T(const T& rhs)
 ----
-   a@ Copy constructor.
 
-a@
-[source]
+_Effects:_ Constructs a [code]#T# instance as a copy of [code]#rhs#.
+
+'''
+
+.[apititle]#Move constructor#
+[source,role=synopsis,id=api:value-semantics-ctor-move]
 ----
-T(T&& rhs);
+T(T&& rhs)
 ----
-   a@ Move constructor.
 
-a@
-[source]
+_Effects:_ Constructs a [code]#T# instance as a move of [code]#rhs#.
+
+'''
+
+.[apititle]#Copy assignment operator#
+[source,role=synopsis,id=api:value-semantics-assignment-copy]
 ----
-T& operator=(const T& rhs);
+T& operator=(const T& rhs)
 ----
-   a@ Copy assignment operator.
 
-a@
-[source]
+_Effects:_ Assigns a copy of [code]#rhs# to this [code]#T# instance.
+
+'''
+
+.[apititle]#Move assignment operator#
+[source,role=synopsis,id=api:value-semantics-assignment-move]
 ----
-T& operator=(T&& rhs);
+T& operator=(T&& rhs)
 ----
-   a@ Move assignment operator.
 
-a@
-[source]
+_Effects:_ Moves from [code]#rhs# and assigns to this [code]#T# instance.
+
+'''
+
+.[apititle]#Destructor#
+[source,role=synopsis,id=api:value-semantics-destructor]
 ----
-~T();
+~T()
 ----
-   a@ Destructor.
 
-|====
+_Effects:_ Destroys this [code]#T# instance.
 
-[[table.hiddenfriends.common.byval]]
-.Common hidden friend functions for by-value semantics
-[width="100%",options="header",separator="@",cols="1,1"]
-|====
-@ Hidden friend function @ Description
-a@
-[source]
+'''
+
+[[sec:value-semantics-hidden-friends]]
+==== Hidden friend functions
+
+This section describes the hidden friend functions that must be provided by all
+classes providing the common by-value semantics.
+Each of these classes may provide further member functions, static member
+functions, etc., as documented in their respective sections.
+
+'''
+
+.[apititle]#Equality operator#
+[source,role=synopsis,id=api:value-semantics-equality]
 ----
 bool operator==(const T& lhs, const T& rhs)
 ----
-   a@ Returns true if this LHS SYCL [code]#T# is equal to the RHS SYCL [code]#T# in accordance with the requirements set out above, otherwise returns false.
 
-a@
-[source]
+_Effects:_ Returns true if [code]#lhs# is equal to [code]#rhs# in accordance
+with the requirements set out above, otherwise returns false.
```

Comment:
Same comment as for reference semantics: I think we should declare the requirements here.